clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ServerGroupController.groovy

```diff
@@ -59,12 +59,27 @@ class ServerGroupController {
 
   @PreAuthorize("hasPermission(#application, 'APPLICATION', 'READ') and hasPermission(#account, 'ACCOUNT', 'READ')")
   @RequestMapping(value = "/applications/{application}/serverGroups/{account}/{region}/{name:.+}", method = RequestMethod.GET)
-  ServerGroup getServerGroup(@PathVariable String application, // needed for @PreAuthorize
+  ServerGroup getServerGroupByApplication(@PathVariable String application, // needed for @PreAuthorize
                              @PathVariable String account,
                              @PathVariable String region,
                              @PathVariable String name) {
+    getServerGroup(account, region, name)
+  }
+
+  @PreAuthorize("hasPermission(#account, 'ACCOUNT', 'READ')")
+  @PostAuthorize("hasPermission(returnObject?.moniker?.app, 'APPLICATION', 'READ')")
+  @RequestMapping(value = "/serverGroups/{account}/{region}/{name:.+}", method = RequestMethod.GET)
```

Comment:
The mixing of `/serverGroups` and `/applications/...` endpoints is starting to have a bad smell. Can you file an issue to have this cleaned up and reference it in a TODO here?

Reply:
- Created issue here: https://github.com/spinnaker/spinnaker/issues/2023
- Added a comment here: https://github.com/spinnaker/clouddriver/pull/1967/files#diff-d8e4115f4785725b1e418532c640c617R62